waf_test.go

```diff
@@ -0,0 +1,263 @@
+package cloudflare
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListWAFPackages(t *testing.T) {
+	setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "GET", r.Method, "Expected method 'GET', got %s", r.Method)
+
+		w.Header().Set("content-type", "application/json")
+		// JSON data from: https://api.cloudflare.com/#zone-analytics-properties
+		fmt.Fprintf(w, `{
+      "success": true,
+      "errors": [],
+      "messages": [],
+      "result": [
+        {
+          "id": "a25a9a7e9c00afc1fb2e0245519d725b",
+          "name": "WordPress rules",
+          "description": "Common WordPress exploit protections",
+          "detection_mode": "traditional",
+          "zone_id": "023e105f4ecef8ad9ca31a8372d0c353",
+          "status": "active"
+        }
+      ],
+      "result_info": {
+        "page": 1,
+        "per_page": 20,
+        "count": 1,
+        "total_count": 2000
+      }
+    }`)
+	}
+
+	mux.HandleFunc("/zones/"+testZoneID+"/firewall/waf/packages", handler)
+
+	want := []WAFPackage{
+		WAFPackage{
+			ID:            "a25a9a7e9c00afc1fb2e0245519d725b",
+			Name:          "WordPress rules",
+			Description:   "Common WordPress exploit protections",
+			ZoneID:        "023e105f4ecef8ad9ca31a8372d0c353",
+			DetectionMode: "traditional",
+			Sensitivity:   "",
+			ActionMode:    "",
+		},
+	}
+
+	d, err := client.ListWAFPackages(testZoneID)
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, d)
+	}
+
+	_, err = client.ListWAFRules(testZoneID, "123")
+	assert.Error(t, err)
+}
+
+func TestListWAFRules(t *testing.T) {
+	setup()
+	defer teardown()
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "GET", r.Method, "Expected method 'GET', got %s", r.Method)
+
+		w.Header().Set("content-type", "application/json")
+		// JSON data from: https://api.cloudflare.com/#zone-analytics-properties
```

Comment:
Same comment as https://github.com/cloudflare/cloudflare-go/pull/210/files#r204581841, it's not matching what is below.